src/core/utils/helpers/general.ts

```diff
@@ -341,8 +365,19 @@ export const getScrollClass = (modalIds: string[]) => {
   return modalIds.length > 0 ? "scroll-lock-background" : "";
 };
 export const dataIsNotEmpty = (data: unknown[]) => Boolean(data.length);
+// Loans with more than warning-threshold days until due
+export const filterLoansNotOverdue = (loans: LoanType[], warning: number) => {
+  return loans.filter(({ dueDate }) => {
+    const due: string = dueDate || "";
+    const daysUntilExpiration = daysBetweenTodayAndDate(due);
+    return daysUntilExpiration - warning > 0;
+  });
+};
 
 export const constructModalId = (prefix: string, fragments: string[]) =>
   `${prefix ? `${prefix}-` : ""}${fragments.join("-")}`;
-
+export const constructSimpleModalId = (
```

Comment:
Why not just using `constructModalId`? I don't see that `constructSimpleModalId ` that is any differnet?

Reply:
Well, as the original need was based on two fixed strings as my modal-id, i thought that calling constructModalId as constructModalId("string1", ["string2"]) was slightly hacky, so i decided that creating a new method, receiving two plain strings as params was a better way to go. 
I removed the new method, and now call the constructModalId as mentioned.